src/Process/LayeredMaterialNodeProcessing.js

```diff
@@ -274,6 +282,10 @@ export function updateLayeredMaterialNodeImagery(context, layer, node) {
                     console.warn(`Imagery texture update error for ${node}: ${err}`);
                 }
                 const definitiveError = node.layerUpdateState[layer.id].errorCount > MAX_RETRY;
+                if (layer.protocol == 'wmts' || layer.protocol == 'wmtsc') {
```

Comment:
No no no :)

Please don't make this code protocol aware.
If remembering the previous `targetLevel` is useful then do it everytime.